internal/services/policy/management_group_policy_assignment_resource.go

```diff
@@ -32,6 +32,7 @@ func (r ManagementGroupAssignmentResource) Arguments() map[string]*pluginsdk.Sch
 			ValidateFunc: validation.All(
 				validation.StringIsNotWhiteSpace,
 				validation.StringDoesNotContainAny("/"),
+				validation.StringLenBetween(3, 24),
```

Comment:
Quickly testing this, it seems the API does accept names of only 1 character like the other `*_policy_assignment` resources, so to avoid breaking existing configurations that may be using that we should change this to 1

```suggestion
				validation.StringLenBetween(1, 24),
```

E.g. this worked:

```hcl
resource "azurerm_management_group_policy_assignment" "test" {
  name                 = "a"
  management_group_id  = azurerm_management_group.test.id
  policy_definition_id = data.azurerm_policy_definition.test.id
  parameters = jsonencode({
    "listOfAllowedLocations" = {
      "value" = [azurerm_resource_group.test.location]
    }
  })
}
```

Could you update the documentation as well?

Reply:
Ah thanks for the docs, was not aware this existed. Updated the length and added the additional validations too.